### PR TITLE
Add Reproject class

### PR DESCRIPTION
**Pull request by larrybradley**
Adds a `Reproject` class to the PSF sandbox for reprojecting pixel coordinates between unrectified and rectified images.

### CODE REVIEW

File: photutils/psf/sandbox.py [hunk 1]
```diff
@@ -304,7 +304,7 @@ class Reproject(object):
 
     Parameters
     ----------
-    wcs_original, wcs_rectified : `~astropy.wcs.WCS` or `~gwcs.wcs.WCS`
+    wcs_original, wcs_rectified : `~astropy.wcs.WCS` or ``gwcs.wcs.WCS``
```
Comment (bsipocz) [inline photutils/psf/sandbox.py]: What about adding gwcs to the intersphinx mapping?

Comment (larrybradley) [inline photutils/psf/sandbox.py]: I don't think this class is exposed in the docs, but that's a better long-term solution.  Done, thanks.

File: .travis.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ env:
         - MAIN_CMD='python setup.py'
         - SETUP_CMD='test'
         - EVENT_TYPE='pull_request push'
-        - PIP_DEPENDENCIES=''
+        - PIP_DEPENDENCIES='gwcs'
```
Comment (bsipocz) [inline .travis.yml]: It doesn't matter much how it's installed being a small package, but we build conda packages for ``gwcs`` in the astropy channel if you ever need it.

Comment (bsipocz) [inline .travis.yml]: good to know. I'll dig up the logs as see what went wrong, there were quite a few issues recently.

Comment (larrybradley) [inline .travis.yml]: Here's the failure:
https://travis-ci.org/astropy/photutils/jobs/209157199#L456

Is gwcs in the `astropy-ci-extras` channel?

Comment (bsipocz) [inline .travis.yml]: no it's in astropy https://anaconda.org/astropy/gwcs/files

Comment (larrybradley) [inline .travis.yml]: So if I add 'astropy` to `CONDA_CHANNELS` it should work?

Comment (bsipocz) [inline .travis.yml]: I really hope so. If it doesn't, then it means another bug upstream.

Comment (larrybradley) [inline .travis.yml]: I'll try that, thanks!

Comment (larrybradley) [inline .travis.yml]: It fails because of conflicting dependencies:
https://travis-ci.org/astropy/photutils/jobs/209395861#L463

Comment (bsipocz) [inline .travis.yml]: oh. Then I'm sorry for the detour, py3.3 indeed doesn't have most of the version combinations.

Comment (larrybradley) [inline .travis.yml]: No worries! 😃

**Pull request merged**